### PR TITLE
update doc to show working example

### PR DESCRIPTION
**Pull request by mkinney**
Resolves: #1634

### CODE REVIEW

**Comment by tianon** [comment]
LGTM, cc @sfackler

File: rust/content.md [hunk 1]
```diff
@@ -39,8 +39,8 @@ COPY . .
 RUN cargo install --path .
 
 FROM debian:buster-slim
-RUN apt-get update && apt-get install -y libssl
-COPY --from=builder /usr/src/myapp /usr/local/bin/myapp
+RUN apt-get update && apt-get install -y libssl-dev
```
Comment (sfackler) [inline rust/content.md]: I don't think the development package is required in this image.

Comment (mkinney) [reply rust/content.md]: What is the non-dev version of libssl-dev? There is no libssl.

Comment (mkinney) [reply rust/content.md]: I think the line is there to show how one could install required packages. A simple myapp probably would not require openssl.

Comment (tianon) [reply rust/content.md]: So maybe both should be replaced by a dummy package name that doesn't exist to further illustrate that it's merely an example?

Comment (sfackler) [reply rust/content.md]: Sorry - `libssl1.1`. Yeah, the line is intended to show how you'd pull in extra runtime dependencies, OpenSSL just being one relatively common example.

Comment (sfackler) [reply rust/content.md]: Sure, we could swap it out to `extra-runtime-dependencies` or something like that.

Comment (mkinney) [reply rust/content.md]: Ok. Makes sense.

File: rust/content.md [hunk 2]
```diff
@@ -39,12 +39,12 @@ COPY . .
 RUN cargo install --path .
 
 FROM debian:buster-slim
-RUN apt-get update && apt-get install -y libssl
-COPY --from=builder /usr/src/myapp /usr/local/bin/myapp
+RUN apt-get update && apt-get install -y extra-runtime-dependencies
+COPY --from=builder /usr/src/myapp/target/release/myapp /usr/local/bin/myapp
```
Comment (elpiel) [inline rust/content.md]: `cargo install` install the binary on the host machine and it doesn't create a release target.
the `myapp` binary will be stored in `$CARGO_HOME/bin/myapp`
`$CARGO_HOME` is an env. variable from the `Dockerfile` of Rust (see here: https://hub.docker.com/layers/rust/library/rust/latest/images/sha256-8b43199e12b23fba6643472c526ce757ad9fe34e89e3485af6cb588b47745959)

E.g. `/home/username/.cargo/bin/validator_worker`

Not sure if it's possible to share an env. variable for use from the builder :thinking:

Comment (mkinney) [reply rust/content.md]: I don't think it is possible to get a variable from one docker multi-stage to the other.

I created a simple hello world to validate values. See https://github.com/mkinney/rust_hello_world

Dockerfile:
```
FROM rust:1.40 as builder
WORKDIR /usr/src/myapp
COPY . .
RUN cargo install --path .

FROM debian:buster-slim
COPY --from=builder /usr/local/cargo/bin/hello_world /usr/local/bin/myapp
CMD ["myapp"]
```

Here is the build output:
```
$ docker build -t myrust/hello .
Sending build context to Docker daemon  1.492MB
Step 1/7 : FROM rust:1.40 as builder
 ---> a18c58410bcf
Step 2/7 : WORKDIR /usr/src/myapp
 ---> Using cache
 ---> ad1e60e2d4a6
Step 3/7 : COPY . .
 ---> dd1317607ed3
Step 4/7 : RUN cargo install --path .
 ---> Running in 08cfb8068905
  Installing hello_world v0.1.0 (/usr/src/myapp)
   Compiling hello_world v0.1.0 (/usr/src/myapp)
    Finished release [optimized] target(s) in 0.19s
  Installing /usr/local/cargo/bin/hello_world
   Installed package `hello_world v0.1.0 (/usr/src/myapp)` (executable `hello_world`)
Removing intermediate container 08cfb8068905
 ---> 2f6313763dbf
Step 5/7 : FROM debian:buster-slim
 ---> 2dbffcb4f093
Step 6/7 : COPY --from=builder /usr/local/cargo/bin/hello_world /usr/local/bin/myapp
 ---> Using cache
 ---> 0c1c6ecaaa90
Step 7/7 : CMD ["myapp"]
 ---> Using cache
 ---> 5a1f47e07eac
Successfully built 5a1f47e07eac
Successfully tagged myrust/hello:latest
```

Here is testing it:
```
$ docker run -it --rm myrust/hello
Hello, world!
```

Comment (elpiel) [reply rust/content.md]: Yeah, that's something I tried as well and it didn't work.
For the example: The only thing I am worried about is `$CARGO_HOME` changing in the Rust docker image, which might break the path if you hardcode it in the other stage.
But that's also good IMO now

**Pull request merged**